### PR TITLE
Adds Img2Img2 Variant for HunyuanDiT.

### PR DESCRIPTION
**Pull request by MnCSSJ4x**
# What does this PR do?
Adds Img2Img variant for HunyuanaDiT as per discussion in #8992. 

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?

How to reproduce:
```python 
import torch
from diffusers import HunyuanDiTPipeline
from diffusers.utils import load_image

pipe = HunyuanDiTImg2ImgPipeline.from_pretrained(
    "Tencent-Hunyuan/HunyuanDiT-Diffusers", torch_dtype=torch.float16
)
pipe.to("cuda")

url = (
    "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/kolors/bunny_source.png"
)
init_image = load_image(url)

# You may also use English prompt as HunyuanDiT supports both English and Chinese
# prompt = "An astronaut riding a horse"
prompt = "一个宇航员在骑马"

image = pipe(prompt, image=init_image).images[0]



```

A colab notebook demonstrating all results can be found [here](). Depth Maps have also been added in the same colab. 

## Who can review?
@a-r-r-o-w @DN6

### CODE REVIEW

**Comment by DN6** [comment]
cc: @yiyixuxu

File: src/diffusers/pipelines/hunyuandit/pipeline_hunyuandit_img2img.py [hunk 1]
```diff
@@ -0,0 +1,1200 @@
...
+        if use_resolution_binning and (height, width) not in SUPPORTED_SHAPE:
+            width, height = map_to_standard_shapes(width, height)
+            height = int(height)
+            width = int(width)
+            logger.warning(
+                f"Reshaped to (height, width)=({height}, {width}), Supported shapes are {SUPPORTED_SHAPE}"
+            )
+
+        # 1. Check inputs. Raise error if not correct
+        self.check_inputs(
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/hunyuandit/pipeline_hunyuandit_img2img.py:955]: can we check `strength` too? https://github.com/huggingface/diffusers/blob/867e0c919e1aa7ef8b03c8eb1460f4f875a683ae/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py#L655

**Review by yiyixuxu — COMMENTED** [review]
looks really good!
thanks for the PR!

can we add a test (similar to https://github.com/huggingface/diffusers/blob/main/tests/pipelines/hunyuan_dit/test_hunyuan_dit.py) and doc too?

File: examples/community/pipeline_hunyuandit_differential_img2img.py [hunk 2]
```diff
@@ -0,0 +1,1242 @@
+# Copyright 2024 HunyuanDiT Authors and The HuggingFace Team. All rights reserved.
```
[the remark this answers is not in the document]
Comment (yiyixuxu) [reply examples/community/pipeline_hunyuandit_differential_img2img.py:1]: oh I realize this pipeline is already in - can we remove this file from this PR then?

**Comment by yiyixuxu** [comment]
@MnCSSJ4x 
The PR is pretty close to merge, I added a "contribution welcome" label so someone in the community can pick it up. Let me know if it is ok. If you have time and would like to work on this, we will remove the label :)

The work left:
1. remove the changes in the example folder, as the community pipeline has already added 
2. add a test 
3. provide a code example and show us some outputs:)

**Comment by vladmandic** [comment]
this is "close-to-merge" for exactly a year now, can diffusers team pick it up and carry over the finish line?
cc @yiyixuxu @DN6

**Comment by vladmandic** [comment]
gentle ping?

**Comment by yiyixuxu** [comment]
hi @vladmandic 
the team really don't have bandwidth for this now.  sorry
